s3-source-connector/src/main/java/io/aiven/kafka/connect/s3/source/utils/AivenS3SourceRecord.java

```diff
@@ -28,20 +28,20 @@
 public class AivenS3SourceRecord {
     private final Map<String, Object> partitionMap;
     private Map<String, Object> offsetMap;
-    private final String toTopic;
+    private final String topic;
     private final Integer topicPartition;
     private final byte[] recordKey;
     private final byte[] recordValue;
 
     private final String objectKey;
 
     public AivenS3SourceRecord(final Map<String, Object> partitionMap, final Map<String, Object> offsetMap,
-            final String toTopic, final Integer topicPartition, final byte[] recordKey, final byte[] recordValue,
+            final String topic, final Integer topicPartition, final byte[] recordKey, final byte[] recordValue,
             final String objectKey) {
         this.partitionMap = new HashMap<>(partitionMap);
         this.offsetMap = new HashMap<>(offsetMap);
 
-        this.toTopic = toTopic;
+        this.topic = topic;
         this.topicPartition = topicPartition;
         this.recordKey = Arrays.copyOf(recordKey, recordKey.length);
         this.recordValue = Arrays.copyOf(recordValue, recordValue.length);
```

Comment:
Why is this copied on the way in, and the way out? Who is mutating this that you need to be cautious of?

Reply:
As such when this record is created in iterator, PMD treats this as it can be mutable and becomes untrusted. Throws EXPOSE-REP2 PMD error during build. 
Instead of Arrays.copy, changed it to clone().

Either we can add this in pmd exception list, or we handle in code.